### PR TITLE
Factor out subtree similarity computations.

### PR DESCRIPTION
**Pull request by snim2**
Simple refactoring of code that isn't much used yet.

### CODE REVIEW

**Comment by snim2** [comment]
Ah, I'll need to fix the spelling in the commit message before merging :(

File: src/lib/similarity.rs [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+              .len() as f64;
+        let common = f64::from(self.num_common_descendants(from, to));
+        let chawathe = common / n_from.max(n_to);
+        debug_assert!(chawathe >= 0. && chawathe <= 1.);
+        chawathe
+    }
+
+    /// Find the number of "common" descendants in two matched subtrees.
+    ///
+    /// To nodes are common if they have already been matched.
```
Comment (ltratt) [inline src/lib/similarity.rs]: "Two"?

File: src/lib/similarity.rs [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        store.push(NodeId::new(1), NodeId::new(3), &Default::default());
+        store.push(NodeId::new(2), NodeId::new(4), &Default::default());
+        assert_eq!(2,
+                   store.num_common_descendants(&NodeId::new(0), &NodeId::new(2)));
+        assert_eq!(2,
+                   store.num_common_descendants(&NodeId::new(0), &NodeId::new(0)));
+        assert_eq!(0,
+                   store.num_common_descendants(&NodeId::new(1), &NodeId::new(0)));
+    }
+
```
Comment (ltratt) [inline src/lib/similarity.rs]: Extraneous blank line.

**Comment by ltratt** [comment]
Is it worth running `rustfmt` on the affected files before commit to get rid of formatting issues?

**Comment by snim2** [comment]
I've been running rustfmt from the text editor, apparently it didn't mind that blank line! Ready to squash and fix the commit message?

**Comment by ltratt** [comment]
Yep, please squash.

**Comment by snim2** [comment]
Squashed

**Comment by snim2** [comment]
Thanks!

**Pull request merged**